src/api/routes/tx.ts

```diff
@@ -80,6 +80,28 @@ export function createTxRouter(db: PgStore): express.Router {
         }
       }
 
+      let fromAddress: string | undefined;
+      if (typeof req.query.from_address === 'string') {
+        if (!isValidC32Address(req.query.from_address)) {
+          throw new InvalidRequestError(
+            `Invalid query parameter for "from_address": "${req.query.from_address}" is not a valid STX address`,
+            InvalidRequestErrorType.invalid_param
+          );
+        }
+        fromAddress = req.query.from_address;
+      }
+
+      let toAddress: string | undefined;
+      if (typeof req.query.to_address === 'string') {
+        if (!isValidC32Address(req.query.to_address)) {
```

Comment:
Recipients can be also be a contract, e.g. https://api.hiro.so/extended/v1/tx/0xcb13a41095cca7f264ab0800d8badf612c54b0e036908d8ebf2d38adc193d2f8

You can use `isValidPrincipal` here instead